### PR TITLE
[WIP] Improvements to SwiftRemoteMirrors

### PR DESCRIPTION
**Pull request by slavapestov**

### CODE REVIEW

**Comment by slavapestov** [comment]
@bitjammer Since this is my first set of changes for the reflection stuff, you mind doing a pre-commit review?

File: CMakeLists.txt [hunk 1]
```diff
@@ -215,6 +215,10 @@ option(SWIFT_STDLIB_ENABLE_RESILIENCE
     "Build the standard libraries and overlays with resilience enabled; see docs/LibraryEvolution.rst"
     FALSE)
 
+option(SWIFT_STDLIB_ENABLE_REFLECTION_METADATA
+    "Build the standard libraries and overlays with remote reflection metadata; see docs/proposals/RemoteMirrors.rst"
+    FALSE)
+
```
Comment (bitjammer) [inline CMakeLists.txt]: Any reason not to go ahead and turn this on now?

**Comment by bitjammer** [comment]
Looks fantastic to me.

The make_shared -> create changes remind me - when we hand over a typeref pointer to the client, which will be opaque to them, they may eventually hand it back asking more questions about it. How do you think we should handle that re: shared pointers vs. a bump pointer allocator or something?

File: CMakeLists.txt (hunk 1 above)
Comment (nadavrot) [inline CMakeLists.txt]: What's the code size impact on libswiftCore.dylib?

**Comment by slavapestov** [comment]
@bitjammer I'm inclined to believe we should switch to a bump allocator and unique TypeRefs, instead of reference counting them. I'd like to use TypeRefs as keys in cache tables (for the struct layout stuff I'm working on).

File: CMakeLists.txt (hunk 1 above)
Comment (slavapestov) [inline CMakeLists.txt]: @bitjammer Well, at the very least we have to shake out all the bugs turning this on might expose. When we turn it on for real, it should be changed in the frontend, not in CMake.

Comment (slavapestov) [inline CMakeLists.txt]: @nadavrot Looks like a modest increase. Note that this is data size, not code size, and it should in theory be true const (but we should check):

    -rwxr-xr-x  1 slava  staff  4298224 Apr 12 15:02 /Users/slava/src/new/build/Ninja-ReleaseAssert+swift-DebugAssert/swift-macosx-x86_64/lib/swift/macosx/libswiftCore.dylib
    -rwxr-xr-x  1 slava  staff  4347376 Apr 12 12:52 /Users/slava/src/new/build/Ninja-ReleaseAssert+swift-DebugAssert/swift-macosx-x86_64/lib/swift/macosx/libswiftCore.dylib

**Comment by slavapestov** [comment]
@swift-ci Please test

**Pull request closed without merging**